src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -377,6 +377,8 @@ public async Task<IActionResult> MoveMediaList(string[] mediaNames, string sourc
                 path = "";
             }
 
+            name = name.Replace('\\', '-').Replace('/', '-'); // Don't create nested folders
```

Comment:
I'm not sure I agree with this change.

I would have thought it creating nested folders was a logical conclusion?

Seems like something we could discuss endlessly :)

Reply:
It does create nested folders on Windows. For example, adding a folder named ´clients/suppliers' works fine (I get a folder 'clients' with another nested folder 'suppliers' inside).

Problem is when I add a folder named 'clients / supplier'. This adds a folder 'clients ' (with a trailing space!) with another nested folder ' suppliers' (with a leading space!) inside.

Folders with leading spaces are no problem, but trailing spaces are! You can't upload anything to it and you can't delete it (not in media library, not in file explorer). You would need a funky command like `rd "\\?\C:\path\media\clients "` to get rid of it.

I could split on `/`, trim each part and join again. Then I thought it probably wasn't the users intention to create nested folders, hence the change above.

Reply:
@deanmarcussen FYI: even in the case when valid nested folders are created (valid = without trailing spaces) the media library doesn't handle this correctly. It shows the deepest folder as a direct descendant of the parent. We should prevent creating nested folders imho.

Reply:
@deanmarcussen You're right!